### ISSUE TITLE
Floating point exception when enabling second hyperion.ng instance

### ISSUE THREAD

**Issue by cmsimike** [hyperion-project/hyperion.ng#936]
- [x] I confirm that this is an issue rather than a question.
This is an issue

## Bug report
When creating a second hyperion instance under "LED Hardware Instance Management"  and starting it, as soon as I configure the second LED hardware, hyperion Ng crashes. Specific exception is Floating point exception (core dumped):
```
[hyperiond MAIN] <INFO> Set user data path to '/home/mike/.hyperion'
[hyperiond DAEMON] <DEBUG> <PythonInit.cpp:45:PythonInit()> Initializing Python interpreter
[hyperiond SettingsManager] <DEBUG> <SettingsManager.cpp:108:SettingsManager()> Settings database initialized
[hyperiond EFFECTFILES] <INFO> 39 effects loaded from directory :/effects/
[hyperiond EFFECTFILES] <INFO> 22 effect schemas loaded from directory :/effects/schema/
[hyperiond EFFECTFILES] <INFO> 0 effects loaded from directory /home/mike/.hyperion/custom-effects
[hyperiond SettingsManager] <DEBUG> <SettingsManager.cpp:108:SettingsManager()> Settings database initialized
[hyperiond BLACKBORDER] <DEBUG> <BlackBorderProcessor.cpp:65:handleSettingsUpdate()> Set mode to: default
[hyperiond ComponentRegister] <DEBUG> <ComponentRegister.cpp:36:setNewComponentState()> Blackborder detector: enabled
[hyperiond SettingsManager] <DEBUG> <SettingsManager.cpp:108:SettingsManager()> Settings database initialized
[hyperiond LEDDEVICE] <INFO> LedDevice 'adalight' found.
[hyperiond ComponentRegister] <DEBUG> <ComponentRegister.cpp:36:setNewComponentState()> Smoothing: enabled
[hyperiond BLACKBORDER] <DEBUG> <BlackBorderProcessor.cpp:65:handleSettingsUpdate()> Set mode to: default
[hyperiond ComponentRegister] <DEBUG> <ComponentRegister.cpp:36:setNewComponentState()> Blackborder detector: enabled
[hyperiond FLATBUFCONNECTION] <INFO> Connecting to Hyperion: 127.0.0.1:19401
[hyperiond EFFECTENGINE] <INFO> Run effect "Rainbow swirl fast" on channel 0
[hyperiond EFFECTENGINE] <DEBUG> <EffectEngine.cpp:183:runEffectScript()> Start the effect: name [Rainbow swirl fast], smoothCfg [2]
[hyperiond HYPERION] <DEBUG> <PriorityMuxer.cpp:158:registerInput()> Register new input 'System/EFFECT' with priority 0 as inactive
[hyperiond HYPERION] <INFO> Initial foreground effect 'Rainbow swirl fast' started
Floating point exception (core dumped)
```

#### Steps to reproduce
1) http://example:8090/#conf_general
2) Create second LED Hardware Instance Management instance
3) save

#### What is expected?
I should be able to control the second device without crashing

#### What is actually happening?
Hyperion is crashing with a floating point exception

#### System

Hyperion Server: 
- Build:       (HEAD detached at f5583de8) (Sersch-f7865b45/f5583de8-1590611619)
- Build time:  May 27 2020 20:36:52
- Git Remote:  https://github.com/hyperion-project/hyperion.ng
- Version:     2.0.0-alpha.6
- UI Lang:     auto (BrowserLang: en-US)
- UI Access:   default
- Avail Capt:  v4l2,framebuffer,x11,qt

Hyperion Server OS: 
- Distribution: Ubuntu 16.04.6 LTS
- Arch:         x86_64
- Kernel:       linux (4.4.0-186-generic (WS: 64))
- Browser:      Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36 

*please note* I cannot run alpha.7 due to https://github.com/hyperion-project/hyperion.ng/issues/903

**Comment by tihoangyeudau** [comment]
https://github.com/hyperion-project/hyperion.ng/issues/903#issuecomment-667749509

This user seem said that problem has been fixed when updating the latest commits. did you try again

**Comment by cmsimike** [comment]
Wow things move so fast here! I only tried alpha.7 a day or two ago. FWIW https://github.com/hyperion-project/hyperion.ng/releases/tag/2.0.0-alpha.7 the changelog here doesn't seem to mention anything similar to an issue I'm having, but I'll build master locally and report back.

**Comment by tihoangyeudau** [comment]
> Wow things move so fast here! I only tried alpha.7 a day or two ago. FWIW https://github.com/hyperion-project/hyperion.ng/releases/tag/2.0.0-alpha.7 the changelog here doesn't seem to mention anything similar to an issue I'm having, but I'll build master locally and report back.

Yes. You need to rebuild with the lastest commit instead of release version, It's outdated

**Comment by cmsimike** [comment]
OK - got it built. Thank you very much for the docker-build. Makes life easy.

So I might have had two different errors. I definitely had one that was zlib, but I also had this one:
>  <ERROR> Device disabled, device 'adalight' signals error: 'Invalid serial device name: [lg_back_lights]!'

which I might have conflated to being related to the zlib issue. alpha.7 still breaks for me (I am using a custom serial path setup through udev). Maybe this should be another issue?

**Comment by cmsimike** [comment]
Here is the udev rule that creates `lg_back_lights`

> SUBSYSTEM=="tty", ATTRS{idVendor}=="16c0", ATTRS{idProduct}=="0483", SYMLINK+="lg_back_lights"

**Comment by m-seker** [comment]
@cmsimike if crash issue is resolved for you then please open another issue for the serial port problem as the title and content of this ticket can be misleading.

**Comment by cmsimike** [comment]
There are a few things here. My original bug of not being able to create a second LED "LED Hardware Instance" which this issue is for. I believe this bug still exists but because I can't run alpha.7 it won't be debugged.

I can't run alpha.7 because of of two separate bugs
1) the zlib bug, which has now been resolved
2) the serial light bug. I can create a new issue for that.

I still believe the original bug in this ticket exists but I will create another ticket for the serial.

**Comment by cmsimike** [comment]
https://github.com/hyperion-project/hyperion.ng/issues/943

**Comment by m-seker** [comment]
I believe the original bug is already resolved but let's wait until you can confirm.

**Comment by cmsimike** [comment]
That would be great! I'll close this as soon as I can confirm. Sorry for all the back and forth on this!

**Comment by cmsimike** [comment]
Floating point exception still happens in latest from master but I have more information.
I am trying to setup and WLED strip. The strip itself is on the bottom of my tv. I set top, left, and right LEDs to 0. Bottom is 150 (since its 5m 30led/m on the bottom of my tv). As soon as I save I get the floating point exception.

```
Hyperion Server: 
- Build:       master (GitHub-2d1a9a2/5a684e2-1598791632)
- Build time:  Aug 30 2020 15:47:36
- Git Remote:  https://github.com/hyperion-project/hyperion.ng.git
- Version:     2.0.0-alpha.7
- UI Lang:     auto (BrowserLang: en-US)
- UI Access:   expert
- Avail Capt:  v4l2,framebuffer,x11,xcb,qt

Hyperion Server OS: 
- Distribution: Ubuntu 16.04.6 LTS
- Arch:         x86_64
- Kernel:       linux (4.4.0-186-generic (WS: 64))
- Browser:      Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.135 Safari/537.36
```

```
Hyperion caught signal :SIGFPE
2020-08-30T09:06:13.473 hyperiond CORE         : <ERROR>        /lib/x86_64-linux-gnu/libpthread.so.0(+0x11390) [0x7f5478cfc390]
2020-08-30T09:06:13.473 hyperiond CORE         : <ERROR>        /usr/bin/hyperiond(hyperion::getLedLayoutGridSize(QJsonArray const&)+0x780) [0x55d397991990]
2020-08-30T09:06:13.473 hyperiond CORE         : <ERROR>        /usr/bin/hyperiond(Hyperion::handleSettingsUpdate(settings::type, QJsonDocument const&)+0x1ec) [0x55d397991f3c]
2020-08-30T09:06:13.474 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QMetaObject::activate(QObject*, int, int, void**)+0x9ae) [0x7f5477d10b1e]
2020-08-30T09:06:13.474 hyperiond CORE         : <ERROR>        /usr/bin/hyperiond(SettingsManager::settingsChanged(settings::type, QJsonDocument const&)+0x33) [0x55d3979b5573]
2020-08-30T09:06:13.474 hyperiond CORE         : <ERROR>        /usr/bin/hyperiond(SettingsManager::saveSettings(QJsonObject, bool)+0xed0) [0x55d3979adb90]
2020-08-30T09:06:13.474 hyperiond CORE         : <ERROR>        /usr/bin/hyperiond(Hyperion::saveSettings(QJsonObject const&, bool)+0x29) [0x55d39798b549]
2020-08-30T09:06:13.474 hyperiond CORE         : <ERROR>        /usr/bin/hyperiond(+0x10f409) [0x55d3979b7409]
2020-08-30T09:06:13.475 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QObject::event(QEvent*)+0xe9) [0x7f5477d11659]
2020-08-30T09:06:13.475 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Widgets.so.5(QApplicationPrivate::notify_helper(QObject*, QEvent*)+0x9c) [0x7f54785c9b8c]
2020-08-30T09:06:13.475 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Widgets.so.5(QApplication::notify(QObject*, QEvent*)+0x261) [0x7f54785d1351]
2020-08-30T09:06:13.475 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QCoreApplication::notifyInternal2(QObject*, QEvent*)+0x110) [0x7f5477ce4c20]
2020-08-30T09:06:13.475 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*)+0x2dd) [0x7f5477ce73bd]
2020-08-30T09:06:13.475 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(+0x2d1cb3) [0x7f5477d38cb3]
2020-08-30T09:06:13.476 hyperiond CORE         : <ERROR>        /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_dispatch+0x2a7) [0x7f5474194197]
2020-08-30T09:06:13.476 hyperiond CORE         : <ERROR>        /lib/x86_64-linux-gnu/libglib-2.0.so.0(+0x4a3f0) [0x7f54741943f0]
2020-08-30T09:06:13.476 hyperiond CORE         : <ERROR>        /lib/x86_64-linux-gnu/libglib-2.0.so.0(g_main_context_iteration+0x2c) [0x7f547419449c]
2020-08-30T09:06:13.476 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)+0x7b) [0x7f5477d390db]
2020-08-30T09:06:13.476 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>)+0xfa) [0x7f5477ce2c0a]
2020-08-30T09:06:13.476 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(QThread::exec()+0xc3) [0x7f5477b11093]
2020-08-30T09:06:13.477 hyperiond CORE         : <ERROR>        /usr/share/hyperion/bin/../lib/libQt5Core.so.5(+0xaed48) [0x7f5477b15d48]
2020-08-30T09:06:13.477 hyperiond CORE         : <ERROR>        /lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba) [0x7f5478cf26ba]
2020-08-30T09:06:13.477 hyperiond CORE         : <ERROR>        /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7f5476a614dd]
Floating point exception (core dumped)
```

**Comment by Lord-Grey** [comment]
The floating point issue for one line of LEDs is already covered by issue #909. Suggest you track that issue until it is resolved.

**Comment by Lord-Grey** [comment]
Closed, as duplicate to #909
